src/sagemaker_containers/entry_point.py

```diff
@@ -0,0 +1,169 @@
+    # type: (str, str, list, dict, bool) -> subprocess.Popen
+    """Runs the user entry point, passing env_vars as environment variables and args as command arguments.
+    If the entry point is:
+        - A Python package: executes the packages as >>> env_vars python -m module_name + args
+        - A Python script: executes the script as >>> env_vars python module_name + args
+        - Any other: executes the command as >>> env_vars /bin/sh -c ./module_name + args
+
+    Example:
+         >>>import sagemaker_containers
+         >>>from sagemaker_containers.beta.framework import entry_point
+
+         >>>env = sagemaker_containers.training_env()
+         {'channel-input-dirs': {'training': '/opt/ml/input/training'}, 'model_dir': '/opt/ml/model', ...}
+
+
+         >>>hyperparameters = env.hyperparameters
+         {'batch-size': 128, 'model_dir': '/opt/ml/model'}
+
+         >>>args = mapping.to_cmd_args(hyperparameters)
+         ['--batch-size', '128', '--model_dir', '/opt/ml/model']
+
+         >>>env_vars = mapping.to_env_vars()
+         ['SAGEMAKER_CHANNELS':'training', 'SAGEMAKER_CHANNEL_TRAINING':'/opt/ml/input/training',
+         'MODEL_DIR':'/opt/ml/model', ...}
+
+         >>>entry_point.run('user_script', args, env_vars)
+         SAGEMAKER_CHANNELS=training SAGEMAKER_CHANNEL_TRAINING=/opt/ml/input/training \
+         SAGEMAKER_MODEL_DIR=/opt/ml/model python -m user_script --batch-size 128 --model_dir /opt/ml/model
+
+     Args:
+        user_entry_point (str): name of the user provided entry point
+        args (list):  A list of program arguments.
+        env_vars (dict): A map containing the environment variables to be written.
+        uri (str): the location of the module.
+        wait (bool): If True, holds the process executing the user entry-point.
+                     If False, returns the process that is executing it.
+     """
+    env_vars = env_vars or {}
+    env_vars = env_vars.copy()
+
+    _files.download_and_extract(uri, user_entry_point, _env.code_dir)
+
+    install(user_entry_point, _env.code_dir)
+
+    _env.write_env_vars(env_vars)
+
+    return call(user_entry_point, args, env_vars, wait)
+
+
+def install(name, dst):
+    """Install the user provided entry point to be executed as follow:
+        - add the path to sys path
+        - if the user entry point is a command, gives exec permissions to the script
+
+    Args:
+        name (str): name of the script or module.
+        dst (str): path to directory with the script or module.
+    """
+    if dst not in sys.path:
+        sys.path.insert(0, dst)
+
+    entrypoint_type = entry_point_type(dst, name)
+    if entrypoint_type is EntryPointType.PYTHON_PACKAGE:
+        _modules.install(dst)
+    if entrypoint_type is EntryPointType.COMMAND:
+        os.chmod(os.path.join(dst, name), 511)
+
+
+def call(user_entry_point, args=None, env_vars=None, wait=True):  # type: (str, list, dict, bool) -> Popen
```

Comment:
I would make this a private method and get rid of the docstring since it's all duplicated from 'run'